tools/wptrunner/wptrunner/executors/protocol.py

```diff
@@ -393,29 +393,32 @@ def switch_to_window(self, wptrunner_id, initial_window=None):
                 if not first or item != initial_window:
                     self.parent.base.set_window(item)
             else:
-                self._switch_to_frame(item)
-
-            try:
-                handle_window_id = self.parent.base.execute_script("return window.__wptrunner_id")
-                if str(handle_window_id) == wptrunner_id:
-                    return
-            except Exception:
-                pass
-            frame_count = self.parent.base.execute_script("return window.length")
-            if frame_count:
-                for frame_id in reversed(range(0, frame_count)):
-                    # None here makes us switch back to the parent after we've processed the frame
-                    stack.append(None)
-                    stack.append(frame_id)
+                try:
+                    self._switch_to_frame(item)
+                except ValueError:
+                    # The frame no longer exists, so continue
+                    pass
+
+            handle_window_id, nested_context_containers = self.parent.base.execute_script("""
```

Comment:
A `try`/`except` block is [required](https://treeherder.mozilla.org/logviewer?job_id=344749984&repo=try&lineNumber=1118) on the `execute_script()` call to handle concurrent DOM manipulation.